src/connection-manager/index.ts

```diff
@@ -688,34 +702,42 @@ export class DefaultConnectionManager extends EventEmitter<ConnectionManagerEven
     )
   }
 
-  async acceptIncomingConnection (maConn: MultiaddrConnection) {
-    try {
-      // check deny list
-      const denyConnection = this.deny.some(ma => {
-        return maConn.remoteAddr.toString().startsWith(ma.toString())
-      })
+  async acceptIncomingConnection (maConn: MultiaddrConnection): Promise<boolean> {
+    // check deny list
+    const denyConnection = this.deny.some(ma => {
+      return maConn.remoteAddr.toString().startsWith(ma.toString())
+    })
 
-      if (denyConnection) {
-        throw errCode(new Error('connection remote address was in deny list'), codes.ERR_CONNECTION_DENIED)
-      }
+    if (denyConnection) {
+      log('connection remote address was in deny list')
+      return false
+    }
 
-      // check allow list
-      const allowConnection = this.allow.some(ma => {
-        return maConn.remoteAddr.toString().startsWith(ma.toString())
-      })
+    // check allow list
+    const allowConnection = this.allow.some(ma => {
+      return maConn.remoteAddr.toString().startsWith(ma.toString())
+    })
 
-      if (allowConnection) {
-        return
-      }
+    if (allowConnection) {
+      return true
+    }
 
-      if (this.getConnections().length < this.opts.maxConnections) {
-        return
+    if (maConn.remoteAddr.isThinWaistAddress()) {
```

Comment:
Is this just checking ip{4,6} + {tcp,udp}? Where does this name come from?

Reply:
> Is this just checking ip{4,6} + {tcp,udp}

Yes.

> Where does this name come from?

I'm actually not sure. I think it refers to just IP + port, no other protocol information - https://www.oilshell.org/cross-ref.html?tag=narrow-waist#narrow-waist